app/controllers/coupons_controller.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+# CouponsController
+class CouponsController < ApplicationController
+  def create
```

Comment:
Add a UseCoupon command: https://github.com/andypike/rectify#commands
And handle different cases like this:
```
on(:ok)      { redirect_to cart_path, notice: t('coupon.added') }
on(:not_exists) { redirect_to cart_path, alert: t('coupon.not_exist') }
on(:already_used) { redirect_to cart_path, alert: t('coupon.used') }
```